consensus/poet/sgx/sawtooth_poet_sgx/libpoet_enclave/poet_enclave.cpp

```diff
@@ -392,6 +394,13 @@ poet_err_t ecall_CreateSignupData(
             ret,
             "Failed to create PoET public/private key pair");
 
+        // Create the monotonic counter bound to the keypair
+        ret =
+            sgx_create_monotonic_counter(
+                &validatorSignupData.counterId,
+                &initialCounterValue);
+        sp::ThrowSgxError(ret, "Failed to create monotonic counter.");
```

Comment:
That's a good point. I have a separate story in Jira to deal with that, but I could roll it into this PR.
Currently counters are destroyed as a consequence of ecall_Terminate which needs to be fixed.
The approach I have in mind is the validator makes a decision when signup data is no longer useful to keep. It should call an enclave method that will release the counter, then the validator needs to clean up its local database of signups. That's a little dicey due to fork handling.